core/src/wallet/cosmos/CosmosLikeMsgFees.hpp

```diff
@@ -0,0 +1,62 @@
+/*
+ *
+ * CosmosLikeMsgFees
+ *
+ * Created by Remi Barjon on 30/03/2020.
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 Ledger
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ */
+
+#ifndef LEDGER_CORE_COSMOSLIKEMSGFEES_H
+#define LEDGER_CORE_COSMOSLIKEMSGFEES_H
+
+#include <api/CosmosLikeAmount.hpp>
+
+#include <string>
+
+namespace ledger {
+namespace core {
+namespace internal {
```

Comment:
Since it's only cosmos related we should put that inside `ledger::core::cosmos` (maybe `internal`, but still inside the cosmos namespace)

Reply:
Basically, we can remove the level of indirection added in `cosmos/cosmos.hpp`, and just define the struct there directly :)

Reply:
Do you mean we should directly create CosmosLikeMsgFees in cosmos.hpp ?

Reply:
yeah I think so, `ledger::core` has no need for the knowledge of `internal::CosmosLikeMsgFees`